pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Here's what the generated patch should look like for the better example above, from the original object stored as an annotation to the new local object:
```
containers:
  -  name: a
     resources:
       requests:
         cpu: 500m
  -  name: b
     env:
      - name: MY_PARAM
      - value: foobar
  -  name: c
     $patch: delete
```
And here's the divergence, from the original object stored as an annotation to the current server object:
```
containers:
  -  name: a
     imagePullPolicy: IfNotPresent
     resources:
       requests:
         cpu: 100m
  -  name: b
     imagePullPolicy: IfNotPresent
     resources:
       requests:
         cpu: 100m
  -  name: c
     imagePullPolicy: IfNotPresent
     resources:
       requests:
         cpu: 100m
```
The divergence conflicts with the patch on the requested cpu for container a, and on the deletion of container c. However, if we ignore additions, then the divergence is empty, since it contains no changes or deletions, and there are no conflicts. Assuming the latter, accepting and applying the patch yields the following new state:
```
containers:
  -  name: a
     image: x
     imagePullPolicy: IfNotPresent
     resources:
       requests:
         cpu: 500m
  -  name: b
     image: y
     imagePullPolicy: IfNotPresent
     resources:
       requests:
         cpu: 100m
     env:
      - name: MY_PARAM
      - value: foobar
```